src/vaadin-grid-sorter.js

```diff
@@ -170,19 +171,34 @@ class GridSorterElement extends ThemableMixin(DirMixin(PolymerElement)) {
   disconnectedCallback() {
     super.disconnectedCallback();
     this._isConnected = false;
+    this._wasDisconnected = true;
   }
 
   /** @private */
-  _pathOrDirectionChanged(path, direction, isConnected) {
-    if (path === undefined || direction === undefined || isConnected === undefined) {
-      return;
-    }
+  _pathOrDirectionChanged() {
+    this.__dispatchSorterChangedEvenIfPossible();
+  }
 
+  /** @private */
+  __isConnectedChanged(isConnected) {
```

Comment:
Does this receive the old value as well? If so the `_wasDisconnected` construct might not be necessary, and you can just check if old and new value differ.

Reply:
Good point. Removed `_wasDisconnected`